kmk/extensions/lock_status.py

```diff
@@ -15,11 +15,12 @@ class LockCode:
 
 class LockStatus(Extension):
     def __init__(self):
-        self.report = 0x00
+        self.report = None
         self.hid = None
         for device in usb_hid.devices:
             if device.usage == HIDUsage.KEYBOARD:
                 self.hid = device
+                self.hid.get_last_received_report()
```

Comment:
You're not even storing the report anywhere. I don't get what this changes.
Even so, `get_last_received_report` is called periodically anyway, why is a call specifically during init (not even during bootup) necessary?

Reply:
Calling `get_last_received_report` here forces `after_hid_send` to be invoked setting the appropriate value for `report` in the class object.

Reply:
Would it make more sense to put a `self.report = self.hid.get_last_received_report` in `during_bootup`?

Reply:
So, my initial code did that. The problem was two fold.
1) I would have had to duplicate my callback code in the init function and I hate duplicating code
2) `after_hid_send` was getting called either way so why not just utilize the code that already exists?

While writing my other comment it dawned on me that with your suggested implementation, we lost the ability to know definitively that report has changed.

Maybe we want another class variable that holds whether or not report was updated on the most recent invocation of `after_hid_send`?